src/NBS.Appointments.Service/Models/AvailabilityHourResponse.cs

```diff
@@ -40,28 +40,33 @@ public static AvailabilityHourResponse FromQflowResponse(SiteSlotsResponse qflow
                 Type = vaccineType
             };
 
-            var availableHours = qflowResponse.Availability
+            var availableSlotTimes = qflowResponse.Availability
                 .Select(x => x)
-                .OrderBy(x => x.Time.Hours)
-                .GroupBy(x => x.Time.Hours)
+                .OrderBy(x => x.Time)
+                .GroupBy(x => x.Time)
                 .Select(x => x.Key)
                 .ToList();
 
             var availabilityByHour = new List<AvailabilityHour>();
 
-            if (!availableHours.Any())
+            if (!availableSlotTimes.Any())
             {
                 response.AvailabilityByHour = availabilityByHour;
                 return response;
             }
 
-            foreach (var hour in availableHours)
+            var currentTime = currentDate.TimeOfDay;
+
+            foreach (var slotTime in availableSlotTimes)
             {
+                if (slotTime < currentTime)
```

Comment:
Would this not exclude things even when the date isn't today? As in it's 1pm on 15/03 but I've asked for appointments on 16/03. Would this not exclude all appointments prior to 1pm?